File: Coding_Questions/JavaScript/SelectionSort.js
// ---Selection Sort---

const selectionSort = (array) => {
  for (let i = 0; i < array.length; i++) {
    let pointer = i;
    for (let c = i + 1; c < array.length; c++) {
      if (array[pointer] > array[c]) {
        pointer = c;
      }
    }
    swap(array, i, pointer);
  }
  return array;
};

const swap = (array, i, j) => {
  const tmp = array[i];
  array[i] = array[j];
  array[j] = tmp;
};

const input = [8, 5, 2, 9, 5, 6, 3];
console.log(selectionSort(input));

File: Coding_Questions/C++/SelectionSort.cpp
// ---Selection Sort---

#include <vector>


std::vector<int> selection_sort(std::vector<int> array) {
    for (int i = 0; i < array.size(); ++i) {
        int pointer = i;
        for (int c = i + 1; c < array.size(); ++c) {
            if (array[pointer] > array[c]) {
                pointer = c;
            }
        }
        std::swap(array[i], array[pointer]);
    }
    return array;
}
